packages/block-library/src/navigation/edit/index.js

```diff
@@ -646,7 +649,8 @@ function Navigation( {
 	// Consider this state as 'unsaved' and offer an uncontrolled version of inner blocks,
 	// that automatically saves the menu as an entity when changes are made to the inner blocks.
 	const hasUnsavedBlocks = hasUncontrolledInnerBlocks && ! isEntityAvailable;
-	if ( hasUnsavedBlocks ) {
+
+	if ( hasUnsavedBlocks && ! isCreatingNavigationMenu ) {
```

Comment:
Addition of `! isCreatingNavigationMenu` is important here. We shouldn't render the uncontrolled inner blocks if a menu is being created. That way we can defer all the UI logic for a "loading" or "saving" state to the main `edit.js` component.